syntaxes/csharp.json

```diff
@@ -123,6 +160,43 @@
         }
       ]
     },
+    "genericConstraints": {
+      "begin": "(where)\\s*(\\w+)\\s*:",
+      "end": "(?={)",
+      "beginCaptures": {
+        "1": {
+          "name": "keyword.other.cs"
+        },
+        "2": {
+          "name": "storage.type.cs"
+        }
+      },
+      "patterns": [
+        {
+          "match": "\\b(class|struct)\\b",
+          "name": "keyword.other.cs"
+        },
+        {
+          "match": "(new)\\s*\\(\\s*\\)",
+          "captures": {
+            "1": {
+              "name": "keyword.other.cs"
+            }
+          }
+        },
+        {
+          "match": "([\\w<>]+)\\s*(?=,|where|{)",
```

Comment:
You are right. I need to think how to handle this better. Whitespace handling is a pain in the a**. :) Will update the pull request.